src/utils.rs

```diff
@@ -41,14 +47,13 @@ pub(crate) fn print_version() {
 /// the user will be repeatedly prompted to provide a valid input
 ///
 /// `error_msg`: will be displayed if user enters an input which does not satisfy the `condition`
-pub(crate) fn get_user_input<F, O, E>(
+pub(crate) fn get_user_input<O, E>(
     prompt: &str,
     default_value: Option<O>,
-    mut predicate: F,
+    condition: fn(input: &str) -> Result<O, E>,
```

Comment:
It will be better to have closure type here.

Reply:
Also as a rule of thumb -- smth is wrong if you use function pointers vs `Fn/FnMut/FnOnce`